services/src/main/java/io/druid/cli/validate/DruidJsonValidator.java

```diff
@@ -69,4 +110,40 @@ public void run()
       throw Throwables.propagate(e);
     }
   }
+
+  private List<String> readData(final StringInputRowParser parser, final CharSource source, final boolean collect)
+      throws IOException
+  {
+    return source.readLines(
+        new LineProcessor<List<String>>()
+        {
+          private final List<String> collected = new ArrayList<String>();
+          private final StringBuilder builder = new StringBuilder();
```

Comment:
Since builder is re-used, this should simply keep making a bigger and bigger string every time `processLine` is called.